### ISSUE TITLE
Multilingual support for rules D400 and D415

### ISSUE THREAD

**Issue by gakugaku** [astral-sh/ruff#16962]
### Overview
The rules [D400](https://docs.astral.sh/ruff/rules/missing-trailing-period/) (missing trailing period) and [D415](https://docs.astral.sh/ruff/rules/missing-terminal-punctuation/) (missing terminal punctuation) in Ruff currently target English only. They do not support terminal punctuation for other languages. For example, a Japanese comment using `。` at the end may be flagged incorrectly.

### Proposed Changes

- Update the logic for D400 and D415 to support multiple languages.  
- Allow configuration of language-specific terminal punctuation (e.g., English: `.`, Japanese: `。`).  

Example for D400/D415 in English:

```python
def average(values: list[float]) -> float:
    """Return the mean of the given values."""
```

Example in Japanese
```python
def average(values: list[float]) -> float:
    """平均値を返します。"""
```

**Comment by MichaReiser** [comment]
I'm somewhat leaning towards using a heuristic here rather than introducing a new configuration option. For example: Allow 。if the first sentence uses any non ascii characters

**Comment by Avasam** [comment]
Why not accept all terminal characters from any language? No heuristic, no configs. Probably low risk of false-negative? (I'd be surprised if someone manages to accidentally mixup terminal characters across languages in a way you'd want the rule to flag)
